internal/git/git.go

```diff
@@ -22,36 +24,48 @@ func CloneCommandFor(s rukpakv1alpha1.GitSource) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return cmd.String(), nil
+	return cmd.String()
 }
 
-func (c *checkoutCmd) String() string {
+func (c *checkoutCmd) String() (string, error) {
 	var checkoutCommand string
 	var repository = c.Repository
 	var directory = c.Directory
 	var branch = c.Ref.Branch
 	var commit = c.Ref.Commit
 	var tag = c.Ref.Tag
+	var preset = ""
+
+	if c.SecretName != "" {
+		repositoryURL, err := url.Parse(repository)
+		if err == nil {
```

Comment:
What happens if we encounter a parsing error here? Does it make sense to silently ignore it?